core/src/parser.rs

```diff
@@ -498,6 +510,66 @@ pub(crate) fn parse_type_alias(t: &ItemType) -> Result<RustItem, ParseError> {
     }))
 }
 
+/// Parses a const variant.
+pub(crate) fn parse_const(c: &ItemConst) -> Result<RustItem, ParseError> {
+    let expr = parse_const_expr(&c.expr)?;
+
+    // serialized_as needs to be supported in case the user wants to use a different type
+    // for the constant variable in a different language
+    let ty = if let Some(ty) = get_serialized_as_type(&c.attrs) {
+        ty.parse()?
+    } else {
+        RustType::try_from(c.ty.as_ref())?
+    };
+
+    match &ty {
+        RustType::Special(SpecialRustType::HashMap(_, _))
+        | RustType::Special(SpecialRustType::Vec(_))
+        | RustType::Special(SpecialRustType::Option(_)) => {
+            return Err(ParseError::RustConstTypeInvalid);
+        }
+        RustType::Special(_s) => (),
+        _ => return Err(ParseError::RustConstTypeInvalid),
+    };
+
+    Ok(RustItem::Const(RustConst {
+        id: get_ident(Some(&c.ident), &c.attrs, &None),
+        r#type: ty,
+        expr,
+    }))
+}
+
+fn parse_const_expr(e: &Expr) -> Result<RustConstExpr, ParseError> {
+    struct ExprLitVisitor(pub Option<Result<RustConstExpr, ParseError>>);
+    impl Visit<'_> for ExprLitVisitor {
+        fn visit_expr_lit(&mut self, el: &ExprLit) {
+            // if self.0.is_some() {
+            //     // should we throw an error instead of silently ignoring a second literal?
+            //     // or would this create false positives?
+            //     return;
+            // }
```

Comment:
Do we want to keep this code or remove it completely? I don't think there's a good reason to keep around commented code.

Reply:
Shoot I commented it while debugging, but it should stay there. Nice catch. 🤦🏻